src/backend/distributed/metadata/node_metadata.c

```diff
@@ -2868,3 +2682,177 @@ UnsetMetadataSyncedForAllWorkers(void)
 
 	return updatedAtLeastOne;
 }
+
+
+/*
+ * SyncDistributedObjects sync the distributed objects to the nodes in metadataSyncContext
+ * with transactional or nontransactional mode according to transactionMode inside
+ * metadataSyncContext.
+ *
+ * Transactions should be ordered like below:
+ * - Nodewide objects (only roles for now),
+ * - Deletion of sequence and shell tables and metadata entries
+ * - All dependencies (e.g., types, schemas, sequences) and all shell distributed
+ *   table and their pg_dist_xx metadata entries
+ * - Inter relation between those shell tables
+ *
+ * Note that we do not create the distributed dependencies on the coordinator
+ * since all the dependencies should be present in the coordinator already.
+ */
+static void
+SyncDistributedObjects(MetadataSyncContext *context)
```

Comment:
Why are these under `node_metadata.c`? Maybe we create a new file `metadata_sync.c` or such, and move related function there?

Reply:
we already have that file. I moved the function there.